### PR TITLE
Added fix to run SQL query function description

### PR DESCRIPTION
**Pull request by wangdatan**
# Description

This change fixed an isseu which the run SQL function cannot correctly summarize the result. In addition to that, this includes a few refactorings to reuse code. And Fixed verbose logic, which correctly shows dataframe when run query with verbose=True

## Type of Change

Please delete options that are not relevant.

- [ ] New Loader/Tool
- [x] Bug fix / Smaller change
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_hub/tools/waii/base.py [hunk 1]
```diff
@@ -34,6 +34,32 @@ def __init__(
         WAII.Database.activate_connection(key=database_key)
         self.verbose = verbose
 
+    def _try_display(self, obj):
+        try:
+            # display df if it is available
+            if self.verbose:
+                display(obj)
+        except:
```
Comment (jerryjliu) [inline llama_hub/tools/waii/base.py:42]: why swallow the try/except here? is there an if/else statement instead?

**Review by jerryjliu — APPROVED** [review]
approving to unblock, other stuff lg

File: llama_hub/tools/waii/base.py (hunk 1 above)
Comment (wangdatan) [reply llama_hub/tools/waii/base.py:42]: The reason is I only want to display the data frame with verbose=True. And if the tool logic runs outside of IPython, there's no `display` function. 

Alternatively, I can check if the function `display` exists or not, maybe that causes less confusion. I can also check if it is inside an IPython environment, but it seems more complex. e.g. https://discourse.jupyter.org/t/find-out-if-my-code-runs-inside-a-notebook-or-jupyter-lab/6935/7 

Let me know if you have any other ideas for this.

Comment (jerryjliu) [reply llama_hub/tools/waii/base.py]: ah i see. ok no problem

**Comment by wangdatan** [comment]
@jerryjliu , I just updated the comments of the `try_display` method to make it less confusing. And I also updated the notebook, which includes the data frame output (tables) to make it easier to understand. And added a notebook block (see the last notebook) to do `chat('run it')` to make sure a query can be executed based on the context.

**Comment by jerryjliu** [comment]
great! i think there's jsut a quick lint error. if you dont' have time to get to this i can do it in a few hours

**Comment by wangdatan** [comment]
Oh, saw it, let me fix it very quickly!

**Comment by wangdatan** [comment]
It is now fixed, @jerryjliu

**Pull request merged**